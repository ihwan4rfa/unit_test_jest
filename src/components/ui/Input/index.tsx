import React from "react";
import { InputProps } from "./types.type";

const Input = (props: InputProps) => {
  const { type, placeholder, label, name } = props;

  return (
    <>
      {label !== "" && <label htmlFor={name}>{label}</label>} <br />
      {type === "textarea" ? (
        <textarea id={name} placeholder={placeholder} />
      ) : (
        <input id={name} type={type} placeholder={placeholder} />
      )}
    </>
  );
};

export default Input;
