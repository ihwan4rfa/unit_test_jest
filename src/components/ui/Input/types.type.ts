export type InputProps = {
  type: string;
  placeholder?: string;
  label: string;
  name: string;
};
