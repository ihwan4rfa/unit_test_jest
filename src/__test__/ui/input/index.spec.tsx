import React from "react";
import { render } from "@testing-library/react";
import Input from "@/components/ui/Input";

describe("input component", () => {
  it("should render input type text", () => {
    const { container, getByLabelText } = render(
      <Input type={"text"} label={"test"} name={"test"} />
    );
    const input = getByLabelText("test");
    expect(input).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("should render inpit type textarea", () => {
    const { container } = render(
      <Input
        type={"textarea"}
        label={"test"}
        placeholder="your commet here"
        name={"test"}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
