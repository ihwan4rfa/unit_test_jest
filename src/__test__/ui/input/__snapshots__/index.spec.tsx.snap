// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input component should render inpit type textarea 1`] = `
<div>
  <label
    for="test"
  >
    test
  </label>
   
  <br />
  <textarea
    id="test"
    placeholder="your commet here"
  />
</div>
`;

exports[`input component should render input type text 1`] = `
<div>
  <label
    for="test"
  >
    test
  </label>
   
  <br />
  <input
    id="test"
    type="text"
  />
</div>
`;
